lib/routes/user/marketplace/vendor_webview.dart

```diff
@@ -64,10 +64,80 @@ class VendorWebViewPageState extends State<VendorWebViewPage> {
     });
   }
 
+  _injectWebLNLibrary() {
+    String includeWebLN = "var webLNLib = document.createElement('script');\n" +
+        "webLNLib.src = 'https://unpkg.com/webln@0.2.0/dist/webln.min.js';\n" +
+        "webLNLib.integrity = 'sha384-mTReBqbhPO7ljQeIoFaD1NYS2KiYMwFJhUNpdwLj+VIuhhjvHQlZ1XpwzAvd93nQ';\n" +
+        "webLNLib.crossOrigin = 'anonymous';\n" +
+        "document.body.appendChild(webLNLib);";
+    _widgetWebview.evalJavascript(includeWebLN).whenComplete(() => _initializeWebLN());
+  }
+
+  _initializeWebLN() {
+    String initWebLN = "   var processedInvoices = [];\n" +
+        "       webln = {\n" +
+        "   enable: function () {\n" +
+        "       window.postMessage(JSON.stringify({ enable: true }));\n" +
+        "       return new Promise(function (resolve, reject) {resolve(true);});\n" +
+        "   },\n" +
+        "   sendPayment: function (paymentRequest) {\n" +
+        "       window.postMessage(JSON.stringify({ uri: paymentRequest }));\n" +
+        "       processedInvoices.push(paymentRequest);\n" +
+        "       return new Promise(function (resolve, reject) { });\n" +
```

Comment:
The standard states that we should either resolve or reject the promise according to if the user has payed successfully or failed/cancelled.
To achieve that I suggest to do the following:
1. Maintain a global mapping of requestID -> pendingPromise (on window object) and register a new request in each sendPayment.
2. In the postMessage of sendPayment add property requestID to the data sent with unique (can be auto-increment) number. 
3. From the webview on getting the sendPayment post message keep the requestID as pending request (We can assume one payment at a time).
4. On  fulfilled payments do something similar to: `_widgetWebview.evalJavascript("window.resolveRequest(requestID)")`
5. On  canceled and failed payments do something similar to: `_widgetWebview.evalJavascript("window.rejectRequest(requestID)")`
6. In javascript map the requestID to the request promise using the global mapping, call either reject or resolve and remove the mapping for this request.